via/via_system.cpp

```diff
@@ -459,13 +459,21 @@ ViaSystem::ViaResults ViaSystem::GenerateInstanceInfo(void) {
         }
     }
 
-    const char* portability_instance_extension_list[] = {VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME};
+    std::vector<const char *> portability_instance_extension_list;
+    inst_info.flags = 0;
+    inst_info.enabledExtensionCount = 0;
+    inst_info.ppEnabledExtensionNames = NULL;
     for (const auto& extension : ext_props) {
         if (strcmp(extension.extensionName, VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0) {
+            portability_instance_extension_list.push_back(VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME);
+            inst_info.ppEnabledExtensionNames = portability_instance_extension_list.data();
             inst_info.flags = VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
-            inst_info.enabledExtensionCount = 1;
-            inst_info.ppEnabledExtensionNames = portability_instance_extension_list;
-            break;
+            inst_info.enabledExtensionCount++;
+        }
+        if (strcmp(extension.extensionName, VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME) == 0) {
```

Comment:
Why did you add device properties 2 here?  I thought this change focused on portability?

Reply:
Maybe a comment might be in order here explaining why...

On Mac, if you specify VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME without specifying VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME, validation produces an error.

Reply:
Not `VK_KHR_portability_enumeration` specifically, but `VK_KHR_portability_subset`  (as the latter is a device extension and needs `vkGetPhysicalDeviceProperties2KHR`).

Reply:
Force-pushed a change adding comments.